core/pkg/server/server.go

```diff
@@ -31,11 +31,11 @@ type Server struct {
 }
 
 // NewServer creates a new server
-func NewServer(ctx context.Context, addr string, portFile string) *Server {
+func NewServer(ctx context.Context, addr string, portFile string) (*Server, error) {
 	listener, err := net.Listen("tcp", addr)
 	if err != nil {
 		slog.Error("can not listen", "error", err)
```

Comment:
instead of logging here, it would be better to return a wrapped error (using `fmt.Errorf`) and let the caller decide whether to log it (i.e. call `slog.Error` in `main` instead)

Reply:
agreed! i just half fixed it...